lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,11 +169,62 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
+
+    // Add Hash.
+    if (ppResult != nullptr && *ppResult != nullptr) {
+      HRESULT HR;
+      (*ppResult)->GetStatus(&HR);
+      if (SUCCEEDED(HR)) {
+        CComPtr<IDxcBlob> pObject;
+        HR = (*ppResult)->GetResult(&pObject);
+        if (SUCCEEDED(HR)) {
+          LPVOID PTR = pObject->GetBufferPointer();
+          if (IsDxilContainerLike(PTR, pObject->GetBufferSize()))
+            HashAndUpdate((DxilContainerHeader *)PTR);
+        }
+      }
+    }
     return S_OK;
   }
   CATCH_CPP_RETURN_HRESULT();
 }
 
+void DxcContainerBuilder::FindHashFunctionFromSource(
+    const DxilContainerHeader *ContainerHeader) {
+  DXASSERT(ContainerHeader != nullptr &&
+               IsDxilContainerLike(ContainerHeader,
+                                   ContainerHeader->ContainerSizeInBytes),
+           "otherwise load function should have returned an error.");
+  static const uint32_t HashStartOffset =
+      offsetof(struct DxilContainerHeader, Version);
+  const BYTE *DataToHash = (const BYTE *)ContainerHeader + HashStartOffset;
+  UINT AmountToHash = ContainerHeader->ContainerSizeInBytes - HashStartOffset;
+  BYTE Result[DxilContainerHashSize];
```

Comment:
It's pre-exiting, but there is no indication that `DxilContainerHashSize` is a constant. I know that some people cringe at Hungarian notation, but the practice in DXC has been to use the k* prefix.

Reply:
Do you mind I change it after this PR with another NFC PR?
Don't want to update other files in current PR.

Reply:
There's multiple constants that this applies to in this file.  This seems like unnecessary churn to me.